core/src/main/java/org/infinispan/notifications/cachelistener/cluster/RemoteClusterListener.java

```diff
@@ -109,14 +109,17 @@ public Address getOwnerAddress() {
          if (trace) {
             log.tracef("Passing Event to manager %s to send to %s", event, origin);
          }
-         eventManager.addEvents(origin, id, Collections.singleton(ClusterEvent.fromEvent(event)), sync);
+         // For non tx events we signal through the current thread as we notify in the same caller for all events
```

Comment:
I can clarify it should really be single notification, not single operation. Because putAll does N notifications. If we add a bulk notification method, we can consolidate them, but that would have to be for another day.